### PR TITLE
Add ClientStage and ClientSessionImpl

### PR DESCRIPTION
**Pull request by bryce-anderson**
The ClientStage is responsible for handling the dispatch of a HTTP
request which means turning a request and response into and from a
stream of HTTP/2 messages.

The ClientSessionImpl is just a thin wrapper around the HttpConnection.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
After this all we need is the `Http2ClientSelector` (probably renamed), and a basic session manager in order to get the first H2 client merged!

File: http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  }
+}
+
+private object ClientStage {
+  private[client] def makeHeaders(request: HttpRequest): Try[Vector[(String, String)]] = {
+    UrlComposition(request.url).map { breakdown =>
+      val hs = new VectorBuilder[(String, String)]
+
+      // h2 pseudo headers
+      hs += StageTools.Method -> request.method.toUpperCase
```
Comment (rossabaker) [inline http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala]: I guess we've ascertained that method doesn't include funny characters, but I still twitch when I see this without a `Locale.ROOT`.

Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala]: We haven't checked the characters, so I'll add a sane `Locale`.

Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala]: `Local.ENGLISH` it is. Thanks for the keen eye.

Comment (rossabaker) [reply http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala]: Somewhere along the way, I picked up `Locale.ROOT` as "the neutral locale", but I can't articulate any difference between the two.  The important thing here is that this isn't broken in Turkey.

**Pull request merged**